app/services/trainees/create_from_dttp.rb

```diff
@@ -0,0 +1,333 @@
+        town_city: dttp_trainee.response["address1_line3"],
+        postcode: dttp_trainee.response["address1_postalcode"],
+        email: dttp_trainee.response["emailaddress1"],
+      }.merge(locale_code)
+    end
+
+    def course_attributes
+      {
+        course_subject_one: course(latest_placement_assignment.response["_dfe_ittsubject1id_value"]),
+        course_subject_two: course(latest_placement_assignment.response["_dfe_ittsubject2id_value"]),
+        course_subject_three: course(latest_placement_assignment.response["_dfe_ittsubject3id_value"]),
+        course_min_age: age_range && age_range[0],
+        course_max_age: age_range && age_range[1],
+        study_mode: study_mode,
+        commencement_date: earliest_placement_assignment.response["dfe_commencementdate"],
+        course_start_date: latest_placement_assignment.response["dfe_programmestartdate"],
+        course_end_date: latest_placement_assignment.response["dfe_programmeeenddate"],
+      }
+    end
+
+    def course(dttp_course_uuid)
+      find_by_entity_id(
+        dttp_course_uuid,
+        Dttp::CodeSets::CourseSubjects::MAPPING,
+      )
+    end
+
+    def age_range
+      @age_range ||= find_by_entity_id(
+        latest_placement_assignment.response["_dfe_coursephaseid_value"],
+        Dttp::CodeSets::AgeRanges::MAPPING,
+      )
+    end
+
+    def study_mode
+      find_by_entity_id(
+        latest_placement_assignment.response["_dfe_studymodeid_value"],
+        Dttp::CodeSets::CourseStudyModes::MAPPING,
+      )
+    end
+
+    def school_attributes
+      return {} if latest_placement_assignment.lead_school_id.blank?
+
+      # Should we raise when schools are not found so that we can add them?
+      attrs = {
+        lead_school: School.find_by(urn: lead_school_urn),
+      }
+
+      if latest_placement_assignment.employing_school_id.present?
+        attrs.merge!({
+          employing_school: School.find_by(urn: employing_school_urn),
+        })
+      end
+
+      attrs
+    end
+
+    def lead_school_urn
+      Dttp::School.find_by(dttp_id: latest_placement_assignment.lead_school_id)&.urn
+    end
+
+    def employing_school_urn
+      Dttp::School.find_by(dttp_id: latest_placement_assignment.employing_school_id)&.urn
+    end
+
+    def funding_attributes
+      return {} unless latest_placement_assignment.response["dfe_allocatedplace"] == Dttp::Params::PlacementAssignment::ALLOCATED_PLACE
+
+      if latest_placement_assignment.response["_dfe_bursarydetailsid_value"] == Dttp::Params::PlacementAssignment::SCHOLARSHIP
+        return { applying_for_scholarship: true }
+      end
+
+      funding_method = FundingMethod.find_by(training_route: training_route_for_funding)
+
+      {
+        applying_for_grant: funding_method&.grant?,
+        applying_for_scholarship: funding_method&.scholarship?,
+        applying_for_bursary: funding_method&.bursary?,
+      }
+    end
+
+    def training_route_for_funding
+      find_by_entity_id(
```

Comment:
If we don't have the mapping in `Dttp::CodeSets::BursaryDetails::MAPPING` this will return nil and funding will quietly be skipped. We should probably flag this up somehow so that we know that we need to add a rule?

We can chat about this all together.

Reply:
Added a fix in https://github.com/DFE-Digital/register-trainee-teachers/pull/1787/commits/3194719a71b08f6b3877aea9dda362c69fb75643